### DISCUSSION TITLE
hexagon-32-qurt-hvx_128 standalone crash without error log

### DISCUSSION THREAD

**Discussion by MagicRabbitTang** [halide/Halide#6141] [Q&A]
Hi,
     i am currently using halide to offload some functions to hexagon dsp. using the standalone mode, with the target hexagon-32-qurt-hvx_128,  when running the lib by the generator, dsp crashs too often......but with no error log,  only crash dump info.
     The example can be seen in qcom hexagon sdk example:device-standalone/conv3x3a16, when comment the following, dsp will crash, but without any useful log to debug.
     I do not know how to debug this case, the only info is the fault pc and the assembly code, but it is hard to debug for the true reason.
#if 0
            bounded_input
                .store_root()
                .compute_at(Func(output), y)
                .align_storage(x, 128)
                .vectorize(x, vector_size, TailStrategy::RoundUp);
#endif

**Comment by aankit-ca** [comment]
Which HexagonSDK are you using?

**Reply by pranavb-ca** [reply]
Also, what is the version of Halide in it?

**Reply by MagicRabbitTang** [reply]
hi，i use hexagon sdk Qualcomm/Hexagon_SDK/3.5.3, the halide lib version is 12.0.0(from our github release). And i have a question what is the difference between the qaulcomm halide lib and the github halide lib.

**Reply by MagicRabbitTang** [reply]
02-27 14:24:05.350 27740 27741 E ./main-conv3x3a16.out: vendor/qcom/proprietary/commonsys-intf/adsprpc/src/fastrpc_apps_user.c:750: Error 0xffffffff: remote_handle_invoke failed for handle 0x3, method 4 on domain 3 (sc 0x4020200)
02-27 14:24:05.350  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: process "/frpc/c05091f0 main-conv3x3a16" crashed in thread "/frpc/c05091f0 " for unknown reason
02-27 14:24:05.350 27740 27741 E ./main-conv3x3a16.out: vendor/qcom/proprietary/commonsys-intf/adsprpc/src/listener_android.c:244:listener protocol failure ffffffff
02-27 14:24:05.351  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: Crashed Shared Object ./libconv3x3a16_skel.so load address : 0x09930000
02-27 14:24:05.351  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: fastrpc_shell_3 load address : E5B00000  and size : D8BF8
02-27 14:24:05.351  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: **__Fault PC   :    0xE5B8738_4*_*
02-27 14:24:05.351  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: LR         :    0x993DF58
02-27 14:24:05.351 27740 27741 E ./main-conv3x3a16.out: vendor/qcom/proprietary/commonsys-intf/adsprpc/src/fastrpc_apps_user.c:750: Error 0x8000040d: remote_handle_invoke failed for handle 0x3, method 4 on domain 3 (sc 0x4020200)
02-27 14:24:05.351 27740 27743 E ./main-conv3x3a16.out: vendor/qcom/proprietary/commonsys-intf/adsprpc/src/fastrpc_apps_user.c:766: Error 0x8000040d: remote_handle64_invoke failed for handle 0x686061e0, method 5 on domain 3 (sc 0x5000100)
02-27 14:24:05.351  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: SP         :    0xF5B81A00
02-27 14:24:05.351 27740 27741 E ./main-conv3x3a16.out: vendor/qcom/proprietary/commonsys-intf/adsprpc/src/listener_android.c:251::error: -2147482611: 0 == (nErr = __QAIC_HEADER(adsp_listener_next2)( ctx, nErr, 0, 0, &ctx, &handle, &sc, inBufs, inBufsLen, &inBufsLenReq))
02-27 14:24:05.351  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: Bad va     :    0xF5B819F8
02-27 14:24:05.351 27740 27741 E ./main-conv3x3a16.out: vendor/qcom/proprietary/commonsys-intf/adsprpc/src/listener_android.c:333:Error 0x8000040d: listener2 thread exited
02-27 14:24:05.351  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: FP         :    0xF5B88228
02-27 14:24:05.351  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: SSR        :    0x0
02-27 14:24:05.351 27740 27743 E ./main-conv3x3a16.out: vendor/qcom/proprietary/commonsys-intf/adsprpc/src/fastrpc_apps_user.c:766: Error 0x8000040d: remote_handle64_invoke failed for handle 0x686061e0, method 5 on domain 3 (sc 0x5000100)
02-27 14:24:05.351  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: **Call trace:
02-27 14:24:05.352  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: [<0993DF58>] conv3x3a16_halide+0x38:     (./libconv3x3a16_skel.so)**
02-27 14:24:05.352  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: [<09935640>] conv3x3a16_run+0x170:     (./libconv3x3a16_skel.so)
02-27 14:24:05.352  1166  1186 V adsprpc : platform_qdi_driver.c:778:0xa60d2:0: [<09934F6C>] conv3x3a16_skel_invoke+0x24C:     (./libconv3x3a16_skel.so)
02-27 14:24:05.352 27740 27743 E ./main-conv3x3a16.out: vendor/qcom/proprietary/commonsys-intf/adsprpc/src/fastrpc_apps_user.c:766: Error 0x8000040d: remote_handle64_invoke failed for handle 0x686061e0, method 5 on domain 3 (sc 0x5000100)

**Reply by MagicRabbitTang** [reply]
hi, the log above is the example crash log. Just crash , no useful log, only __Fault PC : 0xE5B87384..........
